extensions/mongodb-client/runtime/src/main/java/io/quarkus/mongodb/runtime/MongoServiceBindingConverter.java

```diff
@@ -0,0 +1,140 @@
+package io.quarkus.mongodb.runtime;
+
+import static java.lang.String.format;
+
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+
+import org.jboss.logging.Logger;
+
+import io.quarkus.kubernetes.service.binding.runtime.ServiceBinding;
+import io.quarkus.kubernetes.service.binding.runtime.ServiceBindingConfigSource;
+import io.quarkus.kubernetes.service.binding.runtime.ServiceBindingConverter;
+
+/**
+ * ServiceBindingConverter for MongoDB to support SBO (Service Binding Operator) in Quarkus.
+ *
+ * This class supports both the <b>Standard</b> and <b>SRV</b> connection string format for
+ * MongoDB depending on whether port is provided or not.
+ *
+ * <br>
+ * Following individual properties are supported to make the connection string:
+ * <ul>
+ *     <li>username</li>
+ *     <li>password</li>
+ *     <li>host</li>
+ *     <li>port</li>
+ *     <li>database</li>
+ * </ul>
+ * <i>Other than host all other properties are optinoal</i>
+ *
+ * <br>
+ * <br>
+ * Only following options are supported by default:
+ * <ul>
+ * <li>retryWrites=true</li>
+ * <li>w=majority</li>
+ * </ul>
+ *
+ */
+public class MongoServiceBindingConverter implements ServiceBindingConverter {
+    private static final Logger LOGGER = Logger.getLogger(MongoServiceBindingConverter.class);
+
+    private static final String BINDING_TYPE = "mongodb";
+    public static final String BINDING_CONFIG_SOURCE_NAME = BINDING_TYPE + "-k8s-service-binding-source";
+    public static final String MONGO_DB_CONNECTION_STRING = "MONGODB_CONNECTION_STRING";
```

Comment:
This should be the Quarkus property, no?

Reply:
Yes, you're right. I took this from our hack code. The correct property should be `quarkus.mongodb.connection-string`. I'll fix this